src/lib/users/encryption_method.rb

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2016 SUSE LLC.
+#  All Rights Reserved.
+
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of version 2 or 3 of the GNU General
+#  Public License as published by the Free Software Foundation.
+
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+#  GNU General Public License for more details.
+
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+
+#  To contact SUSE about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require "yast"
+Yast.import "UsersSimple"
+
+module Users
+  # Object oriented wrapper to read and set the password encryption method at
+  # Yast::UsersSimple
+  #
+  # @example
+  #   md5 = EncryptionMethod.new("md5") # Raises exception on wrong id
+  #   puts "The system is configured to use #{md5.label}" if md5.current?
+  class EncryptionMethod
+    class Error < RuntimeError
+    end
+
+    class NotFoundError < Error
+    end
+
+    extend Yast::I18n
+    textdomain "users"
+
+    attr_accessor :id
```

Comment:
JFYI it can cause problems in *ancient* ruby before Object#object_id was named Object#id